File: preprocess/readParseData/fn_parseROI.m
function fn_parseROI(mousePath,roiOrder,roiSize)
temp = dir(mousePath);
for k = 1:length(roiOrder)
    mkdir([mousePath filesep 'green_' roiOrder{k}]);
end 

for i = 1:length(temp)     
    tiffPath  = [mousePath filesep temp(i).name];
    fileNames = dir([tiffPath filesep '*.tif']);
    fileName = {fileNames.name};
    
    for j = 1:length(fileName)
        disp(fileName{j})
        try
            fn_saveh5(mousePath,tiffPath,fileName{j},roiOrder,roiSize);
        catch

            disp([fileName{j} ' not done!'])
        end
    end
end 

end

File: pythonPackageRockfish/readParseROI/fn_parseROI.py
import os
from fn_saveh5 import fn_saveh5

def fn_parseROI(mousePath, roiOrder, roiSize):
    for roi in roiOrder:
        os.makedirs(os.path.join(mousePath, f"green_{roi}"), exist_ok=True)

    for sub in os.listdir(mousePath):
        tiffPath = os.path.join(mousePath, sub)
        if not os.path.isdir(tiffPath):
            continue

        for file in os.listdir(tiffPath):
            if not file.endswith(".tif"):
                continue

            print(file)
            try:
                fn_saveh5(mousePath, tiffPath, file, roiOrder, roiSize)
            except Exception as e:
                print(f"{file} not done! ({e})")

